### PR TITLE
Fix graph edges between count-expanded Terraform resources

### PR DESCRIPTION
**Pull request by olivmir**
## Summary
- Fix false positives on graph checks (e.g. `CKV2_AWS_6`) when resources use `count` with a boolean ternary like `var.create_replay_bucket ? 1 : 0`
- When `count` expands a resource, its vertex name becomes indexed (e.g. `aws_s3_bucket.replay[0]`), but edge building stripped `[0]` from references like `aws_s3_bucket.replay[0].id` and looked up `aws_s3_bucket.replay` instead — so the public access block / versioning resources were never connected in the graph
- Parse references with an optional `preserve_resource_index` flag and try indexed vertex lookup before falling back to the stripped name when building edges
## Root cause
Terraform code like this is valid and deploys correctly:
```hcl
resource "aws_s3_bucket" "replay" {
  count = var.create_replay_bucket ? 1 : 0
  ...
}
resource "aws_s3_bucket_public_access_block" "replay" {
  count  = var.create_replay_bucket ? 1 : 0
  bucket = aws_s3_bucket.replay[0].id
  ...
}
```

**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Summary by Aikido
|  Security Issues: 0 |  Quality Issues: 0 |  Resolved Issues: 0 |
| :--- | :--- | :--- |

**⚡ Enhancements**
* Added resource reference lookup variants and normalized quoted index handling
* Expanded edge-case tests and removed unused import from codebase

**🐛 Bugfixes**
* Fixed graph-edge connections for count-expanded Terraform resource references

**🔧 Refactors**
* Precompiled quoted-index regex and avoided redundant lookup calls

<sup>[More info](https://app.aikido.dev/featurebranch/scan/153432199?groupId=6)</sup>

### CODE REVIEW

File: checkov/terraform/graph_builder/local_graph.py [hunk 1]
```diff
@@ -32,6 +32,7 @@
     get_attribute_is_leaf,
     get_referenced_vertices_in_value,
     attribute_has_nested_attributes, remove_index_pattern_from_str,
```
Comment (kapyteinaikido) [inline checkov/terraform/graph_builder/local_graph.py]: `remove_index_pattern_from_str` is no longer needed here? since you moved that logic to utils.py?

**Review by kapyteinaikido — APPROVED** [review]
